itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/reexec/TestReExecuteKilledTezAMQueryPlugin.java

```diff
@@ -195,4 +201,32 @@ private void killAMForQueryId(String queryId) throws Exception {
       }
     }
   }
+
+  boolean isQueryRunning(String queryId, Connection conn) throws Exception {
+    boolean queryRunningInYarn = false;
+    boolean hiveRunning = false;
+
+    // Check YARN state
+    List<ApplicationReport> applicationReports = yarnClient.getApplications();
+    for (ApplicationReport ar : applicationReports) {
+      if (ar.getApplicationTags().contains(queryId) &&
+              ar.getYarnApplicationState() == YarnApplicationState.RUNNING) {
+        queryRunningInYarn = true;
+        break;
+      }
+    }
+    // check if the DAG is also started
+    if (queryRunningInYarn) {
+      try (Statement s = conn.createStatement();
+           ResultSet rs = s.executeQuery(
+                   "SELECT STATE FROM SYS.QUERY WHERE QUERY_ID='" + queryId + "'")) {
+        if (rs.next()) {
+          hiveRunning = "RUNNING".equalsIgnoreCase(rs.getString(1));
+        }
+      } catch (Exception e) {
+        LOG.warn("Exception when checking hive state", e);
+      }
+    }
+    return queryRunningInYarn && hiveRunning;
```

Comment:
minor: `hiveRunning` already incorporates the `queryRunningInYarn` flag since it's inside the if block